src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/SynchronizedMethodHelpers.cs

```diff
@@ -25,13 +25,7 @@ private static void MonitorEnter(object obj, ref bool lockTaken)
                 ObjectHeader.GetLockObject(obj) :
                 SyncTable.GetLockObject(resultOrIndex);
 
-            if (lck.TryAcquire(0))
-            {
-                lockTaken = true;
-                return;
-            }
-
-            Monitor.TryAcquireContended(lck, obj, Timeout.Infinite);
```

Comment:
I've renamed `TryAcquireContended` -> `TryAcquireSlow`. The name was deceiving. This method is not necessarily to handle contention. It is to handle all the scenarios, while quick attempts to acquire can handle only a few common ones.